assets/js/components/DeregistrationModal/DeregistrationModal.test.jsx

```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import { faker } from '@faker-js/faker';
+
+import DeregistrationModal from '.';
+
+describe('Deregistration Modal component', () => {
+  it('should render deregistration modal correctly', async () => {
+    const hostname = faker.name.firstName();
+
+    render(
+      <DeregistrationModal
+        hostname={hostname}
+        isOpen
+        onCleanUp={() => {}}
+        onCancel={() => {}}
+      />
+    );
+
+    expect(await screen.findByText(hostname, { exact: false })).toBeTruthy();
```

Comment:
Curious that we need async waits here

Reply:
Ah that is because the internal `Modal` component uses a transition, so we have to use the async queries to wait for the animation to finish and the elements to appear.